### PR TITLE
Feat/pf tests v2

### PR DESCRIPTION
**Pull request by dividor**
This PR Implements the following:

1. README instructions

2. Addition of Promptflow for LLM output evaluation. Includes a new docker-compose-dev.yml as this wouldn't be needed in production. User needs to build with `docker compose -f docker-compose.yml -f docker-compose-dev.yml up -d --build` to get this functionality

3. Slight refactoring of chainlit ui for neatness

4. Tweak to memory judge to add few-shot examples to better handle context switching. Needed for basic tests

6. A Promptflow wrapper to call chainlit code using Mock chainlit objects to override

NOTE: Though the test can run the same code as chainlit, which will test recipes/memory and on-the-fly assistant as a user would, it has the following limitations:

- No support for images
- Only one test included in flow.dat.json, ie no data.jsonl just yet
- The test function runs, completes, but doesn't exit due to some hanging async process in Chainlit. After a lot of investigation, gave up for now and instead run as a script, then kill the script. VERY hacky workaround.

As noted above this isn't a neat solution due to calling async code of chainlit, but it does work. Will merge into main as a guide for future end-to-end tests, but main focus will be to test recipes server AI and AI assistant in isolation

### CODE REVIEW

File: templates/generate_intent_from_history_prompt.jinja2
Comment (JanPeterDatakind) [inline templates/generate_intent_from_history_prompt.jinja2:1]: Question: The examples' chat history only contains user input and no assistant responses. Didn't we at one point add the assistant output to the history as well for context?

**Review by JanPeterDatakind — APPROVED** [review]
Approving the PR - perhaps we can chat about the comments/ open questions in our next check-in

File: flows/chainlit-ui-evaluation/groundedness_score.jinja2
[the remark this answers is not in the document]
Comment (dividor) [reply flows/chainlit-ui-evaluation/groundedness_score.jinja2:1]: Hmm, good point. I took the prompt from Micsosoft, but no doubt we can improve. As we add tests for LLM-generated output (as opposed to memories/recipes which are deterministic), then we should revisit this.

File: flows/chainlit-ui-evaluation/data.jsonl
[the remark this answers is not in the document]
Comment (dividor) [reply flows/chainlit-ui-evaluation/data.jsonl:1]: That must have been left in there at an earlier point when HAPI gave a slightly different number. Defining the tests will come in the next wave, but I also updated the tests for this PR as I'm extending it to add batch tests. Sorry, I should have started a new branch, more reviewing on this one soon. :)

File: templates/generate_intent_from_history_prompt.jinja2
Comment (dividor) [reply templates/generate_intent_from_history_prompt.jinja2:1]: Well spotted! I've been grappling with this. chat_history is used for intent, and I can see where assistant responses might inform that decision, but the majority comes from the user ...

user: what's the total population of Mali?
assistant: It's XXXXXXX
user: plot a map by state

In terms of intent, this is just as effective I feel, and uses a lot less tokens (bear in mind assistant responses can be BIG) ...

user: what's the total population of Mali?
user: plot a map by state

So I chopped out the AI response. I agree though it is confusing, should we perhaps rename to be chat_history_user or something, or make this more obvious elsewhere?

**Pull request merged**